app/code/Magento/ConfigurableProductGraphQl/etc/graphql/di.xml

```diff
@@ -36,4 +36,7 @@
             </argument>
         </arguments>
     </type>
+    <type name="Magento\ConfigurableProduct\Model\ResourceModel\Attribute\OptionSelectBuilder">
+        <plugin name="configurable_product_graphql_option_select" type="Magento\ConfigurableProductGraphQl\Model\Plugin\AttributeOptionSelect" />
```

Comment:
I think **configurable_product_graphql_limit_disabled_options** will be more meaningful name for the plugin.
Also, please add a order for the plugin.